Source/items.cpp

```diff
@@ -4249,20 +4286,33 @@ void RecreateWitchItem(int ii, int idx, int lvl, int iseed)
 	if (idx == IDI_MANA || idx == IDI_FULLMANA || idx == IDI_PORTAL) {
 		GetItemAttrs(ii, idx, lvl);
 	} else {
+#ifdef HELLFIRE
+		if (idx >= 114 && idx <= 117) {
+			SetRndSeed(iseed);
+			volatile int hi_predelnik = random(0, 1);

```

Comment:
@Predelnik what do you think?

Reply:
I have to look at this closer it's it's unused variable, case is probably a bit different. There are several ways to have effects similar to volatile tbh - using local in inline assembly, uninitialized variable I just couldn't came up with any sane replacement for original volatile.

And now it actually got me into other line of thought that at least first volatile usage might have been to improve debugging experience at some point to avoid variable optimized away which they forgot to remove. Never thought about this technique in actual debugging but it actually might be useful occasionally.

Reply:
Oh that actually sounds plausible, I guess we could then just add a `// BUGFIX remove debugging volatile` and then it's no longer a hack :)